lib/manageiq/session_store.rb

```diff
@@ -0,0 +1,88 @@
+module ManageIQ
+  module SessionStore
+    def self.store
+      @store
+    end
+
+    def self.store=(session_store)
+      adapter = case session_store
+                when :active_record_store then ActiveRecordStoreAdapter.new
+                when :memory_store        then MemoryStoreAdapter.new
+                when :mem_cache_store     then MemCacheStoreAdapter.new
+                else
+                  raise "Please provide a valid Session Store"
+                end
+
+      adapter.patch_session_store_class
+      set_store_config(adapter)
+
+      @store = adapter.class_to_patch.new(nil, adapter.session_options)
+    end
+
+    def self.set_store_config(adapter)
+      Vmdb::Application.config.session_store(*adapter.session_store_config_args)
+      msg = "Using session_store: #{Vmdb::Application.config.session_store}"
+      $log.info("MIQ(SessionStore) #{msg}")
+      puts "** #{msg}" if !Rails.env.production? && adapter.type != :mem_cache_store
+    end
+
+    def self.revoke_all(session_ids)
+      store.delete_sessions(session_ids)
+    end
+
+    class AbstractStoreAdapter
+      ClassToPatch = Class.new
+
+      def self.class_to_patch
+        ClassToPatch
```

Comment:
Is this needed?  Can we instead do:

```ruby
def self.class_to_patch
  raise NotImplementedError, "must be implemented in a subclass"
end
```

Reply:
Yeah, I think I like that better.  I am trying to think if there was a reason for this idea, but I don't think there is.  Thanks.

Reply:
No longer relevant, but still a good suggestion.  +1